common/client.go

```diff
@@ -225,6 +225,12 @@ func (c *DatabricksClient) Authenticate(ctx context.Context) error {
 		c.configureWithGoogleForWorkspace,
 		c.configureWithDatabricksCfg,
 	}
+	// AzureEnvironment could be used in the different contexts, not only for Auzre Authentication
+	azureEnvironment, err := c.getAzureEnvironment()
```

Comment:
Authenticate() might be called after AzureADLSGen2Mount.Config, because of lazy auth design requirements of TF, so this may not fully fix the issue. Configure() is a better place, because it's guaranteed to be called before any resource methods in 100% of cases. Could you add it there along with a line comment referencing the issue?

Also, in scope of this PR, can you replace the lazy loaded calls to azenv with this concrete property?